flink-core/src/test/java/org/apache/flink/api/common/io/FileInputFormatTest.java

```diff
@@ -314,43 +299,95 @@ public void testIgnoredUnderscoreFiles() {
 	}
 
 	@Test
+	public void testExcludeFiles() {
+		try {
+			final String contents = "CONTENTS";
+
+			// create some accepted, some ignored files
+
+			File child1 = temporaryFolder.newFile("dataFile1.txt");
+			File child2 = temporaryFolder.newFile("another_file.bin");
+
+			File[] files = { child1, child2 };
+
+			createTempFiles(contents.getBytes(), files);
+
+			// test that only the valid files are accepted
+
+			Configuration configuration = new Configuration();
+
+			final DummyFileInputFormat format = new DummyFileInputFormat();
+			format.setFilePath(temporaryFolder.getRoot().toURI().toString());
+			format.configure(configuration);
+			format.setFilesFilter(new GlobFilePathFilter(
+				Collections.singletonList("**"),
```

Comment:
* - means any file
** - means any file in any subdirectory 

Added a comment and test for that.